pkg/controller/network/subnet/controller.go

```diff
@@ -132,60 +147,120 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 		cr.SetConditions(runtimev1alpha1.Creating())
```

Comment:
Same as above, it's better to make changes on `status` after the late-initialization `Update` call.